src/prfmodel/stimulus.py

```diff
@@ -113,7 +116,12 @@ class Stimulus:
     # We don't want the object to be hashable because it's mutable
     __hash__ = None  # type: ignore[assignment]
 
-    def __init__(self, design: np.ndarray, grid: np.ndarray, dimension_labels: Sequence[str] | None = None):
+    def __init__(
+        self,
+        design: np.ndarray[Any, np.dtype[np.float32]],
```

Comment:
I would not specify the dtype here and instead for now just propagate the dtype the user has given. We need to rethink dtypes in the entire package anyway (because they need to match across different class instances). The best approach I can think of so far is to add a dtype attribute to every class that converts all tensors/arrays to that dtype, but I would address that in a separate PR (see #41).